src/sentry/preprod/models.py

```diff
@@ -0,0 +1,149 @@
+from typing import Any
+
+from sentry import models
+from sentry.db.models.base import DefaultFieldsModel
+from sentry.db.models.fields.bounded import (
+    BoundedBigIntegerField,
+    BoundedPositiveBigIntegerField,
+    BoundedPositiveIntegerField,
+)
+from sentry.db.models.fields.foreignkey import FlexibleForeignKey
+from sentry.db.models.fields.jsonfield import JSONField
+
+
+class PreprodArtifact(DefaultFieldsModel):
+    """
+    A pre-production artifact provided by the user, presumably from their CI/CD pipeline.
+    With this, we can analyze their artifact and provide them with insights to fix _before_
+    it's released to production.
+
+    Examples:
+    - iOS app builds
+    - Android app builds
+    """
+
+    class ArtifactState:
+        UPLOADING = 0
+        """The user has initiated the upload, but it is not yet complete."""
+        UPLOADED = 1
+        """The upload is complete, but the artifact is not yet processed."""
+        PROCESSING = 2
+        """The artifact is being processed."""
+        PROCESSED = 3
+        """The artifact has been processed and is ready to be used."""
+        FAILED = 4
+        """The artifact failed to process."""
+
+        @classmethod
+        def as_choices(cls):
+            return (
+                (cls.UPLOADING, "uploading"),
+                (cls.UPLOADED, "uploaded"),
+                (cls.PROCESSING, "processing"),
+                (cls.PROCESSED, "processed"),
+                (cls.FAILED, "failed"),
+            )
+
+    class ArtifactType:
+        XCARCHIVE = 0
+        XCFRAMEWORK = 1
```

Comment:
To be extra clear - if we intend for xcarchives/xcframeworks to be zipped like they are in the current state (i.e. .xcarchive.zip), I'd encourage making that explicit in the type naming here.